stubs/Symfony/Component/Form/FormFactoryInterface.stub

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+
+interface FormFactoryInterface
+{
+	/**
+	 * @template TFormType of FormTypeInterface<TData>
+	 * @template TData
+	 *
+	 * @param class-string<TFormType> $type
+	 * @param TData $data
+	 * @param array<string, mixed> $options
+	 *
+	 * @phpstan-return ($data is null ? FormInterface<null|TData> : FormInterface<TData>)
```

Comment:
These complex generics should be tested with TypeInferenceTestCase. To make sure they behave like you expect them to.